pallets/aleph/src/lib.rs

```diff
@@ -121,6 +141,43 @@ pub mod pallet {
         pub(crate) fn set_next_emergency_finalizer(emergency_finalizer: T::AuthorityId) {
             <NextEmergencyFinalizer<T>>::put(emergency_finalizer);
         }
+
+        pub(crate) fn current_session() -> u32 {
+            T::SessionInfoProvider::current_session()
+        }
+
+        pub(crate) fn schedule_next_aleph_bft_version_change(
+            version_change: VersionChange,
+        ) -> Result<(), &'static str> {
+            let current_session = Self::current_session();
+
+            let session_to_schedule = version_change.session;
+            let version_to_schedule = version_change.version_incoming;
+
+            if session_to_schedule < current_session {
+                return Err("Cannot schedule AlephBFT version changes for sessions in the past!");
+            } else if session_to_schedule < current_session + 2 {
+                return Err(
+                    "Tried to schedule AlephBFT version change less than 2 sessions in advance!",
+                );
+            }
+
+            // Can only schedule version changes with a version different than the current one.
+            // If a scheduled future version change is rescheduled to a different session,
+            // it should be possible to reschedule it with the same version as initially.
+            // If a scheduled version change has moved into the past, `SessionManager` records it
+            // as the current version.
+            if let Some(current_version) = <AlephBFTVersion<T>>::get() {
+                if current_version == version_to_schedule {
```

Comment:
in this version we can relax this constraint. We could just allow rescheduling with the same version

Reply:
Uhhh, right, this is tricky. If you want to undo a scheduling of a version change you have to schedule a "change" to the current version, right? So we definitely cannot be erroring out here, unless we add some other way of unscheduling a version change.

Reply:
Wait, did you do _both_? That's somewhat silly, if you want an explicit unschedule then error out on attempts to schedule a change to the same version like before, since then it's an useful error to avoid confusion.

Reply:
After consideration, I got rid of the remover.